pandas/_libs/tslibs/offsets.pyx

```diff
@@ -827,7 +823,55 @@ cpdef int get_day_of_month(datetime other, day_opt) except? -1:
         raise ValueError(day_opt)
 
 
-cpdef int roll_yearday(other, n, month, day_opt='start') except? -1:
+cpdef int _roll_convention(int other, int n, int compare):
+    """
+    Possibly increment or decrement the number of periods to shift
+    based on rollforward/rollbackward conventions.
+
+    Parameters
+    ----------
+    other : int, generally the day component of a datetime
+    n : number of periods to increment, before adjusting for rolling
+    compare : int, generally the day component of a datetime, in the same
+              month as the datetime form which `other` was taken.
```

Comment:
this is very very confusing. you are using this in 2 separate ways. I would prefer 2 functions one for day one for month

Reply:
So is the issue with the docstring in roll_convention?  or the name?  I dont think the function itself can be broken down any further.

Reply:
Would this concern be ameliorated if `roll_monthday` were removed?  I expect all of the places where it is used are eventually going to be simplified/fixed to use `roll_convention` (or something equivalent)

Reply:
I think this is fine, but you are using this two differen types so make
``roll_monthday_interger``, ``roll_monthday_datetime`` or whatever. have 1 function do 1 thing.

Reply:
Is it possible the diff is obfuscating the fact that roll_monthday was split several commits back into `roll_monthday` (which takes datetimes) and `roll_convention` (which takes ints)?